tools/gz_msgs_generate

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import subprocess
+import sys
+
+def main(argv=sys.argv[1:]):
+    parser = argparse.ArgumentParser(
+        description='Generate protobuf support files',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '--protoc-exec',
+        required=True,
+        help='The path to the protoc executable')
+    parser.add_argument(
+        '--gz-generator-bin',
+        help='The path to the gz specific protobuf generator')
+    parser.add_argument(
+        '--generate-cpp',
+        help='Flag to indicate if C++ bindings should be generated')
+    parser.add_argument(
+        '--generate-ruby',
+        help='Flag to indicate if Ruby bindings should be generated')
+    parser.add_argument(
+        '--generate-ignition',
+        help='Flag to indicate if ignition/ headers should be generated')
+    parser.add_argument(
+        '--output-cpp-path',
+        help='The basepath of the generated C++ files')
+    parser.add_argument(
+        '--output-ruby-path',
+        help='The basepath of the generated C++ files')
+    parser.add_argument(
+        '--proto-path',
+        required=True,
+        help='The location of the protos')
+    parser.add_argument(
+        '--input-path',
+        required=True,
+        help='The location of the template files')
+    args = parser.parse_args(argv)
+
+    # First generate the base cpp and ruby files
+    cmd = [args.protoc_exec]
+    cmd += [f'--proto_path={args.proto_path}']
+
+    if args.generate_cpp:
+        cmd += [f'--plugin=protoc-gen-ignmsgs={args.gz_generator_bin}']
+        cmd += [f'--cpp_out=dllexport_decl=IGNITION_MSGS_VISIBLE:{args.output_cpp_path}']
+        cmd += [f'--ignmsgs_out={args.output_cpp_path}']
+    if args.generate_ruby:
+        cmd += [f'--ruby_out=dllexport_decl=IGNITION_MSGS_VISIBLE:{args.output_ruby_path}']
+    cmd += [args.input_path]
+
+    try:
+        subprocess.check_call(cmd)
+    except subprocess.CalledProcessError as e:
+        pass
```

Comment:
Should actual error handling be added before merge?

Reply:
error handling added.